examples/algorithms/plot_metric_closure.py

```diff
@@ -0,0 +1,76 @@
+"""
+==============
+Metric Closure
+==============
+
+The *metric closure* of a graph is the complete graph in which
+each edge weight corresponds to the shortest-path distance
+between two nodes in the original graph.
+
+This example demonstrates the *unweighted* version of the metric closure,
+constructed using :func:`networkx.all_pairs_shortest_path_length`.
+Here, each edge is assumed to have weight 1, so distances correspond to
+the minimum number of hops between nodes.
+
+For weighted graphs, the metric closure can be obtained instead with
+:func:`networkx.all_pairs_dijkstra_path_length`, which computes distances
+using the minimum total edge weights along paths.
+"""
+
+import networkx as nx
+import matplotlib.pyplot as plt
+
+# Creating five nodes from index 0 to 4
+G = nx.Graph()
+G.add_nodes_from(range(5))
+G.add_edge(0, 1)
+G.add_edge(0, 4)
+G.add_edge(1, 3)
+G.add_edge(1, 5)
+G.add_edge(2, 3)
+G.add_edge(2, 4)
+G.add_edge(2, 5)
+G.add_edge(3, 4)
+G.add_edge(3, 5)
+G.add_edge(4, 5)
+
+# Find the all-pairs shortest distance
+# all_pairs_shortest_path_length returns a one-time generator,
+# so wrap with dict() to materialize results for reuse.
+paths = dict(nx.all_pairs_shortest_path_length(G))
+for u, dist_dict in paths.items():
+    print(
+        f"From node {u}, distances to all other nodes: {dist_dict} "
+        "(keys = target nodes, values = shortest-path distances)"
+    )
```

Comment:
I'd just as soon leave this out, or at least simplify the printing with e.g. pprint:

```python
from pprint import pprint
pprint(paths)
```

Again this reduces the amount of "custom" code folks have to read to help focus on the underlying example.

Another minor note: since this is an example I'd vote to be a little more precise with the variable name, i.e. `path_lengths` or `pl` instead of `paths`.